core/sourcehook/sh_memory.h

```diff
@@ -67,27 +67,28 @@ namespace SourceHook
 			// 08048000-0804c000 r-xp 00000000 03:03 1010107    /bin/cat
 			unsigned long rlower, rupper;
 			char r, w, x;
-			while (fscanf(pF, "%lx-%lx %c%c%c", &rlower, &rupper, &r, &w, &x) != EOF) {
+			static char *buffer = NULL;
+			static size_t bufsize = 0;
+			while (getline(&buffer, &bufsize, pF) != -1) {
+				char *addr_split;
+				char *prot_split;
+				rlower = strtoul(buffer, &addr_split, 16);
+				rupper = strtoul(&addr_split[1], &prot_split, 16);
 				// Check whether we're IN THERE!
 				if (laddr >= rlower && laddr < rupper) {
-					fclose(pF);
+					r = prot_split[1];
+					w = prot_split[2];
+					x = prot_split[3];
 					*bits = 0;
 					if (r == 'r')
 						*bits |= SH_MEM_READ;
 					if (w == 'w')
 						*bits |= SH_MEM_WRITE;
 					if (x == 'x')
 						*bits |= SH_MEM_EXEC;
+					fclose(pF);
```

Comment:
probably fine in general, the code size might be smaller if you just set bFound to true here, break, then return bFound at the end but that's anecdotal.

Reply:
I've changed the linux impl to this pattern since it flows better with the non static buffer, should I modify the other impls as well? I can't test them but the CI will I think.

Reply:
Eh, only Linux really matters here.

If there's any more refactoring to do it'd probably be to switch to unique_ptr so we don't have to fclose(), and to find some way to use RAII with getline(). But this is fine as-is, so I'll merge it.